provider/equinix/environ_network.go

```diff
@@ -0,0 +1,199 @@
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+	projectID := e.cloud.Credential.Attributes()["project-id"]
+	ips, _, err := e.equinixClient.ProjectIPs.List(projectID, nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var projectSubnets []network.SubnetInfo
+	for _, ipblock := range ips {
+		subnetID, cidr, err := makeSubnetIDForNetwork(ipblock.ID, ipblock.Network, ipblock.CIDR)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		if ipblock.Facility == nil || ipblock.Facility.Code != e.cloud.Region {
+			continue
+		}
+
+		subnet := network.SubnetInfo{
+			ProviderId:        network.Id(subnetID),
+			ProviderNetworkId: network.Id(ipblock.ID), //TODO: figure out what the network ID should be???
+			CIDR:              cidr,
+			VLANTag:           0,
+			AvailabilityZones: []string{ipblock.Facility.Code},
+		}
+		projectSubnets = append(projectSubnets, subnet)
+	}
+
+	if inst == instance.UnknownId {
+		return projectSubnets, nil
+	}
+
+	device, _, err := e.equinixClient.Devices.Get(string(inst), nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var instanceSubnets []network.SubnetInfo
+nextSubnet: //API client limitation since we can't get the actual blocks for individual instance we have to do this
+	for _, psub := range projectSubnets {
+		_, ipnet, err := net.ParseCIDR(psub.CIDR)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, n := range device.Network {
+			if ipnet.Contains(net.ParseIP(n.Address)) {
+				instanceSubnets = append(instanceSubnets, psub)
+				continue nextSubnet
+			}
+		}
+	}
+
+	return instanceSubnets, nil
+}
+
+func makeSubnetIDForNetwork(networkName, address string, mask int) (string, string, error) {
+	_, netCIDR, err := net.ParseCIDR(fmt.Sprintf("%s/%d", address, mask))
+	if err != nil {
+		return "", "", errors.Annotatef(err, "calculating CIDR for network %q", networkName)
+	}
+
+	cidr := netCIDR.String()
+	subnetID := fmt.Sprintf("subnet-%s", networkName)
+	return subnetID, cidr, nil
+}
+
+// SupportsSpaces returns whether the current environment supports
+// spaces. The returned error satisfies errors.IsNotSupported(),
+// unless a general API failure occurs.
+func (e *environ) SupportsSpaces(context.ProviderCallContext) (bool, error) {
+	return true, nil
+}
+
+// NetworkInterfaces returns a slice with the network interfaces that
+// correspond to the given instance IDs. If no instances where found, but there
+// was no other error, it will return ErrNoInstances. If some but not all of
+// the instances were found, the returned slice will have some nil slots, and
+// an ErrPartialInstances error will be returned.
+func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, ids []instance.Id) ([]network.InterfaceInfos, error) {
+	if len(ids) == 0 {
+		return nil, environs.ErrNoInstances
+	}
+	insts, err := e.Instances(ctx, ids)
+	if err != nil {
```

Comment:
`e.Instances` may return `ErrPartialInstances` if it was able to locate some but not all the instances. I think you should allow the method to proceed in this case (just make sure to check for nils below as suggested in the following comment)